app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormatExpr.scala

```diff
@@ -400,6 +407,87 @@ object Register {
   }
 }
 
+case class CsvColumnName(column: String) extends AnyVal
+
+object CsvColumnName {
+
+  implicit val format: OFormat[CsvColumnName] = derived.oformat()
+
+  val countryCode = "countrycode"
+  val currencyCode = "currencycode"
+  val portCode = "portcode"
+  val portId = "portid"
+  val portType = "porttype"
+  val region = "region"
+}
+
+sealed trait SelectionCriteriaValue
+
+object SelectionCriteriaValue {
+  case class SelectionCriteriaExpr(expr: FormCtx) extends SelectionCriteriaValue
+  case class SelectionCriteriaReference(expr: FormCtx, name: CsvColumnName) extends SelectionCriteriaValue
+  case class SelectionCriteriaSimpleValue(value: List[String]) extends SelectionCriteriaValue
+
+  implicit val formatSelectionCriteriaExpr: OFormat[SelectionCriteriaExpr] = derived.oformat()
+  implicit val formatSelectionCriteriaReference: OFormat[SelectionCriteriaReference] = derived.oformat()
+  implicit val formatSelectionCriteriaSimpleValue: OFormat[SelectionCriteriaSimpleValue] = derived.oformat()
+
+  implicit val format: OFormat[SelectionCriteriaValue] = derived.oformat()
+}
+
+case class SelectionCriteria(column: CsvColumnName, value: SelectionCriteriaValue)
+
+object SelectionCriteria {
+
+  implicit val format: OFormat[SelectionCriteria] = derived.oformat()
+}
+
+case class SimplifiedSelectionCriteria(column: CsvColumnName, value: List[String])
+
+object SimplifiedSelectionCriteria {
+
+  implicit val format: OFormat[SimplifiedSelectionCriteria] = derived.oformat()
+
+  def convertToSimplifiedSelectionCriteria[D <: DataOrigin](
+    lsc: List[SelectionCriteria],
+    lookupRegistry: LookupRegistry,
+    formModelVisibilityOptics: FormModelVisibilityOptics[D]
+  )(
+    implicit
+    l: LangADT
+  ): List[SimplifiedSelectionCriteria] = lsc map {
+    case SelectionCriteria(c, SelectionCriteriaExpr(expr)) =>
+      SimplifiedSelectionCriteria(c, List(formModelVisibilityOptics.eval(expr)))
```

Comment:
Please rebase on top of `master`.

Method `eval` no longer exists. You'll need to use `evalAndApplyTypeInfoFirst` method instead.